### PR TITLE
the implementation of Milestone 2.

### PR DESCRIPTION
**Pull request by jingtaozf**

### CODE REVIEW

File: helm-q.el [hunk 1]
```diff
@@ -1,47 +1,109 @@
...
+   (migemo :initform 'nomultimatch)
+   (volatile :initform t)
+   (nohighlight :initform t)
+   ))
+
+(defun helm-q-instance-display-string (instance)
+  "Argument INSTANCE: one instance."
+  (format "%s\t%s"
+          (cdr (assoc 'address instance))
+          (cdr (assoc 'service instance))))
```
Comment (danielkrizian) [inline helm-q.el]:
```suggestion
(defun helm-q-instance-display-string (instance)
  "Argument INSTANCE: one instance."
  (format "%s %s %s %s"
          (cdr (assoc 'address instance))
          (cdr (assoc 'service instance))
          (cdr (assoc 'env instance))
          (cdr (assoc 'region instance))))
```

File: helm-q.el [hunk 2]
```diff
@@ -1,47 +1,109 @@
...
+
+(defun helm-q-instance-list ()
+  "Load source from json files in a directory."
+  (require 'json)
+  ;; a list whoise members are `(DISPLAY . REAL)' pairs.
```
Comment (danielkrizian) [inline helm-q.el]:
```suggestion
  ;; a list whose members are `(DISPLAY . REAL)' pairs.
```

**Review by danielkrizian — CHANGES_REQUESTED** [review]
It's an excellent first iteration! only two things to look at (plus below two suggested edits):

- [x] 1. [major] from [the mockup](https://docs.google.com/spreadsheets/d/14f2LBU2r__dJkpx5hZBJyuRy94wvKNatSBkw-r8CJhQ/edit#gid=0), can we include also cells B3 / B8:B9 as additional lines of each selection candidate, making each candidate a multiline entity? (e.g. like `helm-org-rifle`). The search pattern would be highlighted in bold within that additional context (see the mockup)

- [x] 2. [minor] a potential bug/unexpected result: try searching for pattern "name" or "namea". the results are unexpected:
UPDATE: seems like fuzzy matching, although the exact behavior is not clear. e.g. "AMALYTICS" does not match "ANALYTICS" despite this string pair seems less dissimilar than "name" vs "emea"
image
image
image

File: helm-q.el (hunk 1 above)
Comment (jingtaozf) [reply helm-q.el]: it's ok.:-)

File: helm-q.el (hunk 2 above)
Comment (jingtaozf) [reply helm-q.el]: Thanks.

**Comment by danielkrizian** [comment]
thanks for the multi-line support. the context is displayed largely as expected. one improvement:

- [ ] [major] when displaying columns of a table, currently the context does not display the associated table:
image
 is it possible to also show table in which the column is defined? it's a hierarchical relationship. one way of displaying that relationship is to collapse the hierarchy using dot `.` separator. The context would then show: ... `Column:'trade.price'`. but please feel free to suggest other possibly better ways

**Comment by jingtaozf** [comment]
I have added table name in the display of column, please update to latest codes.

**Pull request merged**